usersync/syncersbuilder.go

```diff
@@ -26,21 +27,23 @@ func (e SyncerBuildError) Error() string {
 }
 
 func BuildSyncers(hostConfig *config.Configuration, bidderInfos config.BidderInfos) (map[string]Syncer, []error) {
-	// map syncer config by bidder
-	cfgByBidder := make(map[string]config.Syncer, len(bidderInfos))
-	for bidder, cfg := range bidderInfos {
-		if shouldCreateSyncer(cfg) {
-			cfgByBidder[bidder] = *cfg.Syncer
+	biddersWithSyncerCfg := make(map[string]config.BidderInfo, len(bidderInfos))
+	for bidder, bidderInfo := range bidderInfos {
+		if shouldCreateSyncer(bidderInfo) {
+			biddersWithSyncerCfg[bidder] = bidderInfo
 		}
 	}
 
 	// map syncer config by key
 	cfgBySyncerKey := make(map[string][]namedSyncerConfig, len(bidderInfos))
-	for bidder, cfg := range cfgByBidder {
-		if cfg.Key == "" {
-			cfg.Key = bidder
+	for bidder, bidderInfo := range biddersWithSyncerCfg {
+		syncerCopy := *bidderInfo.Syncer
+		var err error
+		syncerCopy.Key, err = getSyncerKey(biddersWithSyncerCfg, bidder, bidderInfo)
+		if err != nil {
+			return nil, []error{err}
```

Comment:
This error line isn't covered by test coverage. Could you update this?

Reply:
nice catch! 
https://github.com/prebid/prebid-server/pull/3082/commits/de6d33a62ec662b6206e0891963bd8e30ed4d5c2 adds test to covers above path